src/java.base/share/classes/java/net/URLEncoder.java

```diff
@@ -222,20 +227,30 @@ public static String encode(String s, String enc)
     public static String encode(String s, Charset charset) {
         Objects.requireNonNull(charset, "charset");
 
-        boolean needToChange = false;
-        StringBuilder out = new StringBuilder(s.length());
+        int i;
+        for (i = 0; i < s.length(); i++) {
+            char c = s.charAt(i);
+            if (c >= 128 || !DONT_NEED_ENCODING.test(c) || c == ' ') {
```

Comment:
```suggestion
            if (!DONT_NEED_ENCODING.test(c) || c == ' ') {
```

I missed this, which is redundant now. Removing this check seems beneficial for mixed or unchanged (linux-x64 workstation):
```
Name                           (unchanged) Cnt  Base   Error   Test   Error  Unit   Diff%
URLEncodeDecode.testEncodeUTF8           0  15 3.575 ± 0.131  3.566 ± 0.158 ms/op    0.2% (p = 0.869 )
URLEncodeDecode.testEncodeUTF8          75  15 1.453 ± 0.051  1.391 ± 0.007 ms/op    4.3% (p = 0.000*)
URLEncodeDecode.testEncodeUTF8         100  15 0.621 ± 0.000  0.542 ± 0.000 ms/op   12.8% (p = 0.000*)
  * = significant
```